# Análise exploratória de dados

In [1]:
import pandas as pd

## 1. Descrição dos dados

In [2]:
df = pd.read_csv('../data/raw/wines_SPA.csv')

In [3]:
df

winery             wine  year  rating  num_reviews  \
0            Teso La Monja            Tinto  2013     4.9           58   
1                   Artadi    Vina El Pison  2018     4.9           31   
2             Vega Sicilia            Unico  2009     4.8         1793   
3             Vega Sicilia            Unico  1999     4.8         1705   
4             Vega Sicilia            Unico  1996     4.8         1309   
...                    ...              ...   ...     ...          ...   
7495               Contino          Reserva  2016     4.2          392   
7496  Conreria d'Scala Dei    Les Brugueres  2018     4.2          390   
7497           Mustiguillo  Finca Terrerazo  2017     4.2          390   
7498           Matarromera     Gran Reserva  2011     4.2          389   
7499              Sei Solo         Preludio  2016     4.2          388   

     country            region   price                  type  body  acidity  
0     Espana              Toro  995.00              Toro Red   5.0      3.0  
1     Espana    Vino de Espana  313.50           Tempranillo   4.0      2.0  
2     Espana  Ribera del Duero  324.95  Ribera Del Duero Red   5.0      3.0  
3     Espana  Ribera del Duero  692.96  Ribera Del Duero Red   5.0      3.0  
4     Espana  Ribera del Duero  778.06  Ribera Del Duero Red   5.0      3.0  
...      ...               ...     ...                   ...   ...      ...  
7495  Espana             Rioja   19.98             Rioja Red   4.0      3.0  
7496  Espana          Priorato   16.76           Priorat Red   4.0      3.0  
7497  Espana      El Terrerazo   24.45                   Red   4.0      3.0  
7498  Espana  Ribera del Duero   64.50  Ribera Del Duero Red   5.0      3.0  
7499  Espana  Ribera del Duero   31.63  Ribera Del Duero Red   5.0      3.0  

[7500 rows x 11 columns]

In [4]:
df_dict = pd.DataFrame([
    {
        'variavel':'winery',
        'descricao':'Vinícola: local onde o vinho foi produzido',
        'tipo':'qualitativa',
        'subtipo':'nominal'
    },{
        'variavel':'nome do wine',
        'descricao':'vinho',
        'tipo':'qualitativa',
        'subtipo':'nominal'
    },{
        'variavel':'year',
        'descricao':'ano',
        'tipo':'quantitativa',
        'subtipo':'discreta'
    },{
        'variavel':'rating',
        'descricao':'avaliação',
        'tipo':'quantitativa',
        'subtipo':'discreta'
    },{
        'variavel':'num_reviews',
        'descricao':'número de avaliações',
        'tipo':'quantitativa',
        'subtipo':'discreta'
    },{
        'variavel':'country',
        'descricao':'país de origem',
        'tipo':'qualitativa',
        'subtipo':'nominal'
    },{
        'variavel':'region',
        'descricao':'região do vinho',
        'tipo':'qualitativa',
        'subtipo':'nominal'
    },{
        'variavel':'price',
        'descricao':'preço em euros',
        'tipo':'quantitativa',
        'subtipo':'continua'
    },{
        'variavel':'type',
        'descricao':'tipo do vinho',
        'tipo':'qualitativa',
        'subtipo':'nominal'
    },{
        'variavel':'body',
        'descricao':'corpo do vinho: como o vinho "cai" na boca',
        'tipo':'quantitativa',
        'subtipo':'discreta'
    },{
        'variavel':'acidity',
        'descricao':'acidez',
        'tipo':'quantitativa',
        'subtipo':'discreta'
    }
])

In [5]:
df_dict

variavel                                   descricao          tipo  \
0         winery  Vinícola: local onde o vinho foi produzido   qualitativa   
1   nome do wine                                       vinho   qualitativa   
2           year                                         ano  quantitativa   
3         rating                                   avaliação  quantitativa   
4    num_reviews                        número de avaliações  quantitativa   
5        country                              país de origem   qualitativa   
6         region                             região do vinho   qualitativa   
7          price                              preço em euros  quantitativa   
8           type                               tipo do vinho   qualitativa   
9           body  corpo do vinho: como o vinho "cai" na boca  quantitativa   
10       acidity                                      acidez  quantitativa   

     subtipo  
0    nominal  
1    nominal  
2   discreta  
3   discreta  
4   discreta  
5    nominal  
6    nominal  
7   continua  
8    nominal  
9   discreta  
10  discreta

- São 11 variáveis, sendo 1 quantitativa contínuas, 5 quantitativas discretas e 5 qualitativas nominais.

In [6]:
df_dict.to_csv('../data/external/dict.csv', index=False)

## 2. Perguntas de partida e hipóteses

In [7]:
number_of_na_by_column = df.isnull().sum().loc[lambda x: x > 0]
number_of_na_rows = df.isnull().any(axis=1).sum()

In [8]:
print(df.isnull().sum())
number_of_na_rows = df.isnull().any(axis=1).sum()
number_of_na_by_column = df.isnull().sum().loc[lambda x: x > 0]

print(f'''
A partir dessa informação percebemos que existem valores faltantes na(s) variável(is) {number_of_na_by_column.axes[0].tolist()}.
Além disso, sabemos que existem {number_of_na_rows} linhas com valores faltantes''')

winery            0
wine              0
year              2
rating            0
num_reviews       0
country           0
region            0
price             0
type            545
body           1169
acidity        1169
dtype: int64

A partir dessa informação percebemos que existem valores faltantes na(s) variável(is) ['year', 'type', 'body', 'acidity'].
Além disso, sabemos que existem 1171 linhas com valores faltantes


In [9]:
df.describe(include='all')

winery     wine  year       rating   num_reviews country region  \
count      7500     7500  7498  7500.000000   7500.000000    7500   7500   
unique      480      847    71          NaN           NaN       1     76   
top     Contino  Reserva  2011          NaN           NaN  Espana  Rioja   
freq        457      467  1190          NaN           NaN    7500   2440   
mean        NaN      NaN   NaN     4.254933    451.109067     NaN    NaN   
std         NaN      NaN   NaN     0.118029    723.001856     NaN    NaN   
min         NaN      NaN   NaN     4.200000     25.000000     NaN    NaN   
25%         NaN      NaN   NaN     4.200000    389.000000     NaN    NaN   
50%         NaN      NaN   NaN     4.200000    404.000000     NaN    NaN   
75%         NaN      NaN   NaN     4.200000    415.000000     NaN    NaN   
max         NaN      NaN   NaN     4.900000  32624.000000     NaN    NaN   

              price       type         body      acidity  
count   7500.000000       6955  6331.000000  6331.000000  
unique          NaN         21          NaN          NaN  
top             NaN  Rioja Red          NaN          NaN  
freq            NaN       2357          NaN          NaN  
mean      60.095822        NaN     4.158427     2.946612  
std      150.356676        NaN     0.583352     0.248202  
min        4.990000        NaN     2.000000     1.000000  
25%       18.900000        NaN     4.000000     3.000000  
50%       28.530000        NaN     4.000000     3.000000  
75%       51.350000        NaN     5.000000     3.000000  
max     3119.080000        NaN     5.000000     3.000000

- Como só existe um país, Espanha, podemos retirá-lo para a continuação das observções.

In [10]:
del df['country']

In [11]:
df

winery             wine  year  rating  num_reviews  \
0            Teso La Monja            Tinto  2013     4.9           58   
1                   Artadi    Vina El Pison  2018     4.9           31   
2             Vega Sicilia            Unico  2009     4.8         1793   
3             Vega Sicilia            Unico  1999     4.8         1705   
4             Vega Sicilia            Unico  1996     4.8         1309   
...                    ...              ...   ...     ...          ...   
7495               Contino          Reserva  2016     4.2          392   
7496  Conreria d'Scala Dei    Les Brugueres  2018     4.2          390   
7497           Mustiguillo  Finca Terrerazo  2017     4.2          390   
7498           Matarromera     Gran Reserva  2011     4.2          389   
7499              Sei Solo         Preludio  2016     4.2          388   

                region   price                  type  body  acidity  
0                 Toro  995.00              Toro Red   5.0      3.0  
1       Vino de Espana  313.50           Tempranillo   4.0      2.0  
2     Ribera del Duero  324.95  Ribera Del Duero Red   5.0      3.0  
3     Ribera del Duero  692.96  Ribera Del Duero Red   5.0      3.0  
4     Ribera del Duero  778.06  Ribera Del Duero Red   5.0      3.0  
...                ...     ...                   ...   ...      ...  
7495             Rioja   19.98             Rioja Red   4.0      3.0  
7496          Priorato   16.76           Priorat Red   4.0      3.0  
7497      El Terrerazo   24.45                   Red   4.0      3.0  
7498  Ribera del Duero   64.50  Ribera Del Duero Red   5.0      3.0  
7499  Ribera del Duero   31.63  Ribera Del Duero Red   5.0      3.0  

[7500 rows x 10 columns]

In [12]:
df.describe(include='all')

winery     wine  year       rating   num_reviews region        price  \
count      7500     7500  7498  7500.000000   7500.000000   7500  7500.000000   
unique      480      847    71          NaN           NaN     76          NaN   
top     Contino  Reserva  2011          NaN           NaN  Rioja          NaN   
freq        457      467  1190          NaN           NaN   2440          NaN   
mean        NaN      NaN   NaN     4.254933    451.109067    NaN    60.095822   
std         NaN      NaN   NaN     0.118029    723.001856    NaN   150.356676   
min         NaN      NaN   NaN     4.200000     25.000000    NaN     4.990000   
25%         NaN      NaN   NaN     4.200000    389.000000    NaN    18.900000   
50%         NaN      NaN   NaN     4.200000    404.000000    NaN    28.530000   
75%         NaN      NaN   NaN     4.200000    415.000000    NaN    51.350000   
max         NaN      NaN   NaN     4.900000  32624.000000    NaN  3119.080000   

             type         body      acidity  
count        6955  6331.000000  6331.000000  
unique         21          NaN          NaN  
top     Rioja Red          NaN          NaN  
freq         2357          NaN          NaN  
mean          NaN     4.158427     2.946612  
std           NaN     0.583352     0.248202  
min           NaN     2.000000     1.000000  
25%           NaN     4.000000     3.000000  
50%           NaN     4.000000     3.000000  
75%           NaN     5.000000     3.000000  
max           NaN     5.000000     3.000000

- Dos 7500 totais, são únicos 847 vinhos, 480 vinícolas, 76 regiões e 21 tipos.
- O vinho mais comum é o Reserva com 467 aparições.
- A vinícola mais comum é a Contino com 457 aparições.
- O ano mais comum é 2011 com 1190 aparições.
- A região mais comum é Rioja com 2440 aparições.
- O tipo mais comum é Rioja Red com 2357 aparições.
- A média de preço é, aproximadamente, 60.10 euros.

## 3. Insights